stake-pool/cli/src/main.rs

```diff
@@ -1758,8 +2058,36 @@ fn main() {
                     .help("Public key for the new stake pool staker."),
             )
         )
+        .subcommand(SubCommand::with_name("set-sol-deposit-authority")
+            .about("Change sol deposit authority account for the stake pool. Must be signed by the manager.")
+            .arg(
+                Arg::with_name("pool")
+                    .index(1)
+                    .validator(is_pubkey)
+                    .value_name("POOL_ADDRESS")
+                    .takes_value(true)
+                    .required(true)
+                    .help("Stake pool address."),
+            )
+            .arg(
+                Arg::with_name("new_sol_deposit_authority")
+                    .index(2)
+                    .validator(|x| {
+                        #[allow(clippy::cmp_owned)]
+                        let default = if x == String::from("none") {
+                            Ok(())
+                        } else {
+                            Err(String::from("Not a pubkey or 'none'"))
+                        };
+                        is_pubkey(x).or(default)
+                    })
+                    .value_name("ADDRESS_OR_NONE")
+                    .takes_value(true)
+                    .help("'none', or a public key for the new stake pool sol deposit authority."),
```

Comment:
How about following the convention from the preferred validator CLI?  There, you specify `--unset` if you want to unset or `--vote-account <PUBKEY_HERE>` to specify the preferred validator vote account.  I'm not sure which is better, but we should at least be consistent

Reply:
Ok, I prefer `--set-to-none`, and have the two options `conflicts_with` one another

Reply:
Sure! Check out where `--unset` is being used, and rename all of those too then.  It works exactly in that way